src/parser.rs

```diff
@@ -1243,17 +1260,52 @@ impl<'input, 'abbrev, 'unit, Endian> DebuggingInformationEntry<'input, 'abbrev,
         }
     }
 
-    /// Find the first attribute in this entry which has the given name,
-    /// and return its value. Returns `Ok(None)` if no attribute is found.
-    pub fn attr_value(&self, name: constants::DwAt) -> Option<AttributeValue<'input>> {
+    fn attr_value_internal<F>(&self, name: constants::DwAt, func: F) -> Option<AttributeValue<'input>>
+        where F: Fn(constants::DwAt, AttributeValue<'input>) -> AttributeValue<'input>
+    {
         let mut attrs = self.attrs();
         while let Ok(Some(attr)) = attrs.next() {
             if attr.name() == name {
-                return Some(attr.value());
+                return Some(func(name, attr.value()));
             }
         }
         None
     }
+
+    /// Run some common fixups to present DWARF attributes in more useful forms.
+    fn prettify_attr_value(name: constants::DwAt, value: AttributeValue<'input>) -> AttributeValue<'input> {
+        match name {
+            constants::DW_AT_stmt_list => {
+                let offset = DebugLineOffset(match value {
+                    AttributeValue::Data(data) if data.len() == 4 => {
+                        Endian::read_u32(data) as u64
+                    }
+                    AttributeValue::Data(data) if data.len() == 8 => {
+                        Endian::read_u64(data)
+                    }
+                    AttributeValue::SecOffset(offset) => offset,
+                    otherwise => return otherwise,
+                });
+                AttributeValue::DebugLineRef(offset)
+            },
+            _ => value,
+        }
+    }
+
+    /// Find the first attribute in this entry which has the given name,
+    /// and return its value, without any attempt to clean it up. Returns
+    /// `Ok(None)` if no attribute is found.
+    pub fn attr_value_raw(&self, name: constants::DwAt) -> Option<AttributeValue<'input>> {
+        self.attr_value_internal(name, |_, x| x)
+    }
+
+    /// Find the first attribute in this entry which has the given name,
+    /// and return its value, after running `prettify_attr_value` on it.
+    /// Returns `Ok(None)` if no attribute is found.
+    pub fn attr_value(&self, name: constants::DwAt) -> Option<AttributeValue<'input>> {
+        self.attr_value_internal(name,
+                                 DebuggingInformationEntry::<'input, 'abbrev, 'unit, Endian>::prettify_attr_value)
```

Comment:
Nit: instead of making the `_internal` variant take a callback, do

```rust
self.attr_value_raw(...).map(|val| self.prettify_attr_value(name, val));
```

and then you can get rid of the `_internal` variant completely by folding it into the `_raw` version.